oktapam/data_source_cloud_connection.go

```diff
@@ -0,0 +1,80 @@
+package oktapam
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+)
+
+func dataSourceCloudConnection() *schema.Resource {
+	return &schema.Resource{
+		Description: descriptions.SourceCloudConnection,
+		ReadContext: dataSourceCloudConnectionFetch,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
isnt the `ID` `Computed: true` ?
and `Name` not `Computed: true` ?

Reply:
I'm not an expert on this but I don't think so. From the existing examples the `ID` is always required on the data source and the other fields are marked as `computed`. I think this because Terraform will use this `ID` to uniquely identify the resource, and it will retrieve the values for those computed fields from the resource using the provided ID.

Reply:
the data-source is basically the tf file that we write?
or it is the internal storage where when the tf queries the system it holds that value as data-source?

Reply:
No, the data source defines two things:
1. The entity data schema
2. Methods so Terraform can operate on that entity: create, read, update, delete

The purpose of the code we can write in a`.tf` file (or in the tests like in this PR changes) is to execute that terraform code we wrote to manage a resource.

Reply:
> isnt the `ID` `Computed: true` ? and `Name` not `Computed: true` ?

All fields in a data source definition has to be marked as `computed` except the ID. Because the purpose of a data source is to read existing data.

Reply:
got it
i thought it was the reverse.
thanks for the clarification!